accepted/2020/windows-specific-apis/windows-specific-apis.md

```diff
@@ -0,0 +1,402 @@
+
+In some cases, we get away with marking enum members as platform-specific. If
+the API is popular enough, we may want to provide a custom analyzer that checks
+for certain arguments.
+
+## Open Issues
+
+* We should run the tool again .NET 5
+* We currently don't have a way to mark an API as unsupported on a given
+  platform. That is, the design is agnostic (no attribute) or inclusion list
+  (attributes with support platforms). We need to add this feature to support
+  Blazor.
+    - `System.Security.Cryptography.OpenSSL` should be marked as not supported
+      on Windows
+* We should consider special casing enum values in the analyzer and only
+  complain about assignments/passing to parameters). We can probably get away
+  with marking the constructors of the types that override with platform
+  specific implementations.
+* We should consider how marking virtual methods work in the analyzer.
+* We need to find a good code base for running this analyzer on as
+  dotnet/runtime, dotnet/roslyn are mostly not consuming platform-specific APIs.
+
+## Entire assemblies
+
+There is a set of assemblies that we can mark wholesale as being
+Windows-specific:
+
+> **OPEN**: Should `System.Threading.Overlapped.dll` be removed?
+
+* **Microsoft.Win32.Registry.dll**
+* **Microsoft.Win32.Registry.AccessControl.dll**
+* **Microsoft.Win32.SystemEvents.dll**
+* **System.Diagnostics.EventLog.dll**
+* **System.Diagnostics.PerformanceCounter.dll**
+* **System.DirectoryServices.dll**
+* **System.DirectoryServices.AccountManagement.dll**
+* **System.IO.FileSystem.AccessControl.dll**
+* **System.IO.Pipes.AccessControl.dll**
+* **System.Management.dll**
+* **System.Net.Http.WinHttpHandler.dll**
+* **System.Security.AccessControl.dll**
+* **System.Security.Cryptography.Cng.dll**
+* **System.Security.Cryptography.ProtectedData.dll**
+* **System.Security.Principal.Windows.dll**
+* **System.ServiceProcess.ServiceController.dll**
+* **System.Threading.AccessControl.dll**
+* **System.Threading.Overlapped.dll**
+
+## Specific APIs
+
+### System
+
+> **OPEN** Should we treat those as Windows-only or should we consider them as
+> partially portable?
+
+* **Console**
+    - `Beep(Int32, Int32)`
+    - `get_CapsLock()`
+    - `get_CursorVisible()`
+    - `get_NumberLock()`
+    - `get_Title()`
+    - `MoveBufferArea(Int32, Int32, Int32, Int32, Int32, Int32)`
+    - `MoveBufferArea(Int32, Int32, Int32, Int32, Int32, Int32, Char, ConsoleColor, ConsoleColor)`
+    - `set_BufferHeight(Int32)`
+    - `set_BufferWidth(Int32)`
+    - `set_CursorSize(Int32)`
+    - `set_WindowHeight(Int32)`
+    - `set_WindowLeft(Int32)`
+    - `set_WindowTop(Int32)`
+    - `set_WindowWidth(Int32)`
+    - `SetBufferSize(Int32, Int32)`
+    - `SetWindowPosition(Int32, Int32)`
+    - `SetWindowSize(Int32, Int32)`
+
+### System.Configuration
+
+* **DpapiProtectedConfigurationProvider**
+
+### System.Diagnostics
+
+* **Process**
+    - `EnterDebugMode()`
+    - `LeaveDebugMode()`
```

Comment:
these two are `nops` and to be consistent with other nops, we should remove them from the list

context: https://github.com/dotnet/designs/pull/138/files#r454448405

```suggestion
```